aws/rust-runtime/aws-config/src/lib.rs

```diff
@@ -376,63 +397,10 @@ mod loader {
     }
 }
 
-mod connector {
-
-    // create a default connector given the currently enabled cargo features.
-    // rustls  | native tls | result
-    // -----------------------------
-    // yes     | yes        | rustls
-    // yes     | no         | rustls
-    // no      | yes        | native_tls
-    // no      | no         | no default
-
-    use crate::provider_config::HttpSettings;
-    use aws_smithy_async::rt::sleep::AsyncSleep;
-    use aws_smithy_client::erase::DynConnector;
-    use std::sync::Arc;
-
-    // unused when all crate features are disabled
-    #[allow(dead_code)]
-    pub(crate) fn expect_connector(connector: Option<DynConnector>) -> DynConnector {
-        connector.expect("A connector was not available. Either set a custom connector or enable the `rustls` and `native-tls` crate features.")
-    }
-
-    #[cfg(any(feature = "rustls", feature = "native-tls"))]
-    fn base(
-        settings: &HttpSettings,
-        sleep: Option<Arc<dyn AsyncSleep>>,
-    ) -> aws_smithy_client::hyper_ext::Builder {
-        let mut hyper =
-            aws_smithy_client::hyper_ext::Adapter::builder().timeout(&settings.timeout_settings);
-        if let Some(sleep) = sleep {
-            hyper = hyper.sleep_impl(sleep);
-        }
-        hyper
-    }
-
-    #[cfg(feature = "rustls")]
-    pub(crate) fn default_connector(
-        settings: &HttpSettings,
-        sleep: Option<Arc<dyn AsyncSleep>>,
-    ) -> Option<DynConnector> {
-        let hyper = base(settings, sleep).build(aws_smithy_client::conns::https());
-        Some(DynConnector::new(hyper))
-    }
-
-    #[cfg(all(not(feature = "rustls"), feature = "native-tls"))]
-    pub(crate) fn default_connector(
-        settings: &HttpSettings,
-        sleep: Option<Arc<dyn AsyncSleep>>,
-    ) -> Option<DynConnector> {
-        let hyper = base(settings, sleep).build(aws_smithy_client::conns::native_tls());
-        Some(DynConnector::new(hyper))
-    }
-
-    #[cfg(not(any(feature = "rustls", feature = "native-tls")))]
-    pub(crate) fn default_connector(
-        _settings: &HttpSettings,
-        _sleep: Option<Arc<dyn AsyncSleep>>,
-    ) -> Option<DynConnector> {
-        None
-    }
+// unused when all crate features are disabled
+/// Unwrap an [`Option<DynConnector>`](aws_smithy_client::erase::DynConnector), and panic with a helpful error message if it's `None`
+pub fn expect_connector(
```

Comment:
This was private to the crate before, and I think it can still be that way from what I see in the diff. If it remains private, then its organization doesn't matter as much.
```suggestion
pub(crate) fn expect_connector(
```

Reply:
Since I moved the default connector stuff back into aws-config, I put those and `expect_connector` in a new `aws_config::connector` module. I then added the `pub(crate)` you suggested.